framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -858,7 +859,22 @@ private void parseField(
         // StubParser parses all annotations in type annotation position as type annotations
         annotateDecl(declAnnos, elt, decl.getElementType().getAnnotations());
         AnnotatedTypeMirror fieldType = atypeFactory.fromElement(elt);
-        annotate(fieldType, decl.getElementType());
+
+        VariableDeclarator fieldVarDecl = null;
+        for (VariableDeclarator var : decl.getVariables()) {
+            if (var.getName().toString().equals(elt.getSimpleName().toString())) {
+                fieldVarDecl = var;
+            }
+        }
+        assert fieldVarDecl != null;
+
+        annotate(fieldType, fieldVarDecl.getType());
+
+        if (fieldType.getKind() == TypeKind.ARRAY) {
+            annotate(((AnnotatedArrayType) fieldType).getComponentType(), decl.getAnnotations());
```

Comment:
Shouldn't this go to the component type that is no longer an array?
I.e. in `@A int [] [] f` I think the current code attaches the annotation on the nested array, instead of on the actual component type. I would have expected a loop here (or maybe we already have a helper for this...).
If that happens in `annotate`, a comment would be helpful.